### PR TITLE
Escape key now closes Add Funds dialog

### LINKED ISSUE #3800: Escape key not closing add funds dialog
**Did you search for similar issues before submitting this one?**
Searched for "esc"

**Describe the issue you encountered:**
Once you open the "Add funds..." dialog on the Payments page you can't close it using just the keyboard. This affects Accessibility and general UI/UX. I see that the dialog component has a listener for it so I'm assuming something else is catching that keyboard event.

**Expected behavior:**

Hitting escape should close a dialog modal.
- Platform (Win7, 8, 10? macOS? Linux distro?):
  macOS
- Brave Version:
  0.12.0
- Steps to reproduce:
  1. Go to preferences > payments
  2. Click on "Add Funds..."
  3. Hit the escape key
- Screenshot if needed:
- Any related issues:

### PR DESCRIPTION
**Pull request by jasonrsadler**
Resolves #3800 

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:

1. - Enable payments
2. - Click 'Add Funds'
3. - Press escape key to verify 'Add Funds' dialog closes
---
1. - Enable payments
2. - Click 'Add Funds'
3. - Choose a currency
4. - Press escape key to verify 'Add Funds' dialog closes
5. - Click 'Add Funds' and verify that dialog begins at first step

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by jasonrsadler** [comment]
I agree. It's still a wip and there's something else going on in sync that isn't allowing it to work. Also, it's losing focus on some of the subdialogs if you click the button to go back and then try escape (example go to add funds, pick a currency, then click previous and esc no longer works until you click somewhere on the page.

**Comment by jasonrsadler** [comment]
Escape key should now work with all the payments area.

The Sync area is partially implemented with some areas losing focus.

**Comment by NejcZdovc** [comment]
@jasonrsadler that's fine

File: app/common/state/aboutPreferencesState.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (NejcZdovc) [inline app/common/state/aboutPreferencesState.js]: I think this file is not needed here

**Comment by NejcZdovc** [comment]
@jasonrsadler it's working now. The only thing that is not working now is

1. clean profile
2. click on add funds
3. click next
4. press ESC

**Comment by jasonrsadler** [comment]
Yes. I left out the intro screen. I'll make sure it's included.

**Comment by NejcZdovc** [comment]
master 3d3ec33
0.23 eb5de93f9
0.22 6b53911df

**Pull request merged**